lib/reservation_manager.rb

```diff
@@ -0,0 +1,143 @@
+require "pry"
+require "date"
+
+class Reservation_Manager
+  attr_reader :rooms, :all_reservations, :available_rooms
+  attr_accessor :all_block_reservations
+
+  def initialize
+    @all_reservations = all_reservations || []
+    @all_block_reservations = all_block_reservations || []
+  end
+
+  def make_reservation(check_in, check_out, is_hotel_blocker: false, block_id: nil)
+    fixed_check_in = Date.parse(check_in)
+    fixed_check_out = Date.parse(check_out)
+
+    if is_hotel_blocker == true
+      reservation = Reservation.new(block_id, is_hotel_blocker: true)
+
+      all_block_reservations.each do |block_reservation|
+        if block_id == block_reservation.reservation_id
+          reservation.check_in = block_reservation.check_in
+          reservation.check_out = block_reservation.check_out
+        end
+      end
+
+      rooms_for_block = all_block_reservations.map { |block_instance| block_instance.room }
+      reservation.room = rooms_for_block[0]
+      all_block_reservations.shift
+    else
+      reservation = Reservation.new(1)
+      reservation.check_in = fixed_check_in
+      reservation.check_out = fixed_check_out
+
+      available_rooms = find_available_rooms(check_in, check_out)
+      if available_rooms.length > 0
+        reservation.room = available_rooms[0]
+      else
+        raise ArgumentError, "There are no available rooms at the moment. Please try again later!"
+      end
+    end
+
+    all_reservations << reservation
+    return reservation
+  end
+
+  def make_hotel_block(block_id, check_in, check_out, room_num)
```

Comment:
I'd like to see the `Reservation_Manager` handle `block_id`s for the user.  That way a user can't accidentally give out the same `block_id` twice.